### PR TITLE
Carrying #26863 — Remove deprecated `:` separator for security option

### PR DESCRIPTION
**Pull request by vdemeester**
Carrying #26863 — with versioning the change.

I tried to keep the "version" knowledge in `api/server` and thus making other validation changes that would need to be versionned easier to write.

- [ ] Might need an integration-test though 👼 

/cc @thaJeztah @cpuguy83 @jhowardmsft @justincormack @AkihiroSuda 

Signed-off-by: Vincent Demeester <vincent@sbr.pm>

### CODE REVIEW

File: api/server/router/container/container_routes.go [hunk 1]
```diff
@@ -380,6 +382,15 @@ func (s *containerRouter) postContainersCreate(ctx context.Context, w http.Respo
 		HostConfig:       hostConfig,
 		NetworkingConfig: networkingConfig,
 		AdjustCPUShares:  adjustCPUShares,
+	}, func(config container.Config, hostConfig container.HostConfig) error {
+		if versions.GreaterThanOrEqualTo(version, "1.27") {
+			for _, opt := range hostConfig.SecurityOpt {
+				if !strings.Contains(opt, "=") {
```
Comment (justincormack) [inline api/server/router/container/container_routes.go]: Thats not right, `--security-opt no-new-privileges` is also valid.

Comment (vdemeester) [inline api/server/router/container/container_routes.go]: Ah dang it 😅 I'll have to do something else then 😓

**Comment by vdemeester** [comment]
@justincormack updated 👼

**Comment by vdemeester** [comment]
(oh the CI jobs didn't run though.. don't merge until they did)

File: daemon/daemon_unix.go [hunk 2]
```diff
@@ -177,6 +177,10 @@ func parseSecurityOpt(container *container.Container, config *containertypes.Hos
 		if strings.Contains(opt, "=") {
 			con = strings.SplitN(opt, "=", 2)
 		} else if strings.Contains(opt, ":") {
+			// Note: keeping this to be retro-compatible. From API 1.27 and later, a validation is
+			// done upfront and this will fail early.
+			// But as long as we support API versions lower than 1.27, we should still be
+			// able to handle `:` at this point of the code
```
Comment (thaJeztah) [inline daemon/daemon_unix.go]: IIUC, these options are read from the container's HostConfig, which means that a container _created_ with an old API version still would have `:`. Should we

- _migrate_ the options in the container-config?
- start thinking of a more standardized way of storing these options?

**Comment by justincormack** [comment]
Needs a rebase

**Comment by tiborvass** [comment]
Ping @thaJeztah

**Comment by thaJeztah** [comment]
We may want to move this to the next _quarterly_ release, depending on the outcome of https://github.com/docker/docker/pull/31848#discussion_r106158749

**Comment by vdemeester** [comment]
Good point. I think I can close this for now then 👼

**Pull request closed without merging**